src/org/ods/component/Pipeline.groovy

```diff
@@ -151,8 +151,17 @@ class Pipeline implements Serializable {
 
                         if (!registry.get(NexusService)) {
                             logger.debug 'Registering NexusService'
-                            registry.add(NexusService, new NexusService(
-                                context.nexusUrl, context.nexusUsername, context.nexusPassword))
+                            script.withCredentials([
+                                script.usernamePassword(
+                                    credentialsId: context.credentialsId,
+                                    usernameVariable: 'USERNAME',
+                                    passwordVariable: 'PASSWORD'
+                                )
+                            ]) {
+                                NexusService nexusService = new NexusService(context.nexusUrl,
+                                    script.env.USERNAME as String, script.env.PASSWORD as String)
```

Comment:
It is not a good practice to store clear-text credentials in long-living attributes. It would be better to use withCredentials inside the NexusService whenever needed. Is there anything that prevents us from doing this?

Reply:
This was my first approach when I started coding it, but I found the blocker of CPS methods defined here https://www.jenkins.io/doc/book/pipeline/cps-method-mismatches/. Some of the methods in the NexusService class that needed credentials were already labeled as @NonCPS and when I introduced the credentials there the methods needed to be CPS per definition of Jenkins. When I removed the label of the methods, this created a cascade of errors which I could not solve as it impacted methods way above that were already labeled as @NonCPS and needed to be changed too. I lost a lot of time trying to fix it but the task complicatedtoo much and needed more of a general refactor of the shared library. This is the reason why i decided the other approach which was simpler and cleaner.